changelog/fragments/go-v3-default.yaml

```diff
@@ -0,0 +1,28 @@
+entries:
+  - description: >
+      Go CLI plugin go/v3 is now the default.
+    kind: change
+    breaking: false
+    migration:
+      header: Upgrade your Go project from "go.kubebuilder.io/v2" to "go.kubebuilder.io/v3"
+      body: >
+        The newly released go/v3 plugin has many new features and (breaking) changes
+        incompatible with projects created by go/v2. You are not required to upgrade
+        and your go/v2 project will continue to work with new operator-sdk versions.
+        If you wish to upgrade, check out the upstream
+        [migration guide](https://master.book.kubebuilder.io/migration/plugin/plugins.html).
```

Comment:
IMO we need to add here to update the layout from `layout: go.kubebuilder.io/v2` to `layout: go.kubebuilder.io/v3-alpha`
Then, I think would be better the kb link at the end for users are able to look for further info.

Reply:
I'd rather defer completely to upstream docs for this migration, since you may have to make some other changes before upgrading your project config file's `layout` key.